pkg/controller/s3/bucket/websiteConfig.go

```diff
@@ -139,28 +139,37 @@ func GenerateWebsiteConfiguration(config *v1beta1.WebsiteConfiguration) *awss3.W
 	if config.RedirectAllRequestsTo != nil {
 		wi.RedirectAllRequestsTo = &awss3.RedirectAllRequestsTo{
 			HostName: awsclient.String(config.RedirectAllRequestsTo.HostName),
-			Protocol: awss3.Protocol(config.RedirectAllRequestsTo.Protocol),
+		}
+		if config.RedirectAllRequestsTo.Protocol != "" {
+			wi.RedirectAllRequestsTo.Protocol = awss3.Protocol(config.RedirectAllRequestsTo.Protocol)
 		}
 	}
-	wi.RoutingRules = make([]awss3.RoutingRule, len(config.RoutingRules))
-	for i, rule := range config.RoutingRules {
-		rr := awss3.RoutingRule{
-			Redirect: &awss3.Redirect{
+	if config.RoutingRules != nil {
+		wi.RoutingRules = make([]awss3.RoutingRule, len(config.RoutingRules))
+		for i, rule := range config.RoutingRules {
+			redirect := &awss3.Redirect{
```

Comment:
Why did you need to switch from initializing with `RoutingRule` to `Redirect`? The former seems more straightforward to read to me.

Reply:
i did this becaue i wasn't 100% sure if i could do `rr.Redirect.Protocol` but i realise now that i can do that (i'm a bit rusty in Go as i haven't written it in ages)

Reply:
updated